eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/HttpPipelinePolicyCheck.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Http Pipeline Policy Checks
+ * Any class that implements the HttpPipelinePolicy interface should:
+ *  <ol>
+ *    <li>must be a public class</li>
+ *    <li>not in an implementation package or sub-package</li>
+ *  </ol>
+ */
+public class HttpPipelinePolicyCheck extends AbstractCheck {
+    private static final String HTTP_PIPELINE_POLICY = "HttpPipelinePolicy";
+    private boolean isImplePackage;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.PACKAGE_DEF,
+            TokenTypes.CLASS_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        switch (ast.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                String packageName = FullIdent.createFullIdentBelow(ast).getText();
+                isImplePackage = packageName.contains("implementation");
+                break;
+            case TokenTypes.CLASS_DEF:
+                String className = ast.findFirstToken(TokenTypes.IDENT).getText();
+                // Get all interfaces name
+                DetailAST impleClauseToken = ast.findFirstToken(TokenTypes.IMPLEMENTS_CLAUSE);
+                // Skip check if the class doesn't implement any interface
+                if (impleClauseToken == null) {
+                    return;
+                }
+                Set<String> interfaces = new HashSet<>();
+                for (DetailAST node = impleClauseToken.getFirstChild(); node != null; node = node.getNextSibling()) {
```

Comment:
You can update this to use the TokenUtil for simplicity
```
TokenUtil.forEachChild(impleClauseToken, TokenTypes.IDENT, node -> interfaces.add(node .getText());
```

Reply:
outdated